### PR TITLE
color grading workflow

### PR DESCRIPTION
**Pull request by HogJonny-AMZN**
LUT resources/assets, python scripts, and test data for Nuke and PhotoShop.

Relates to generating LUTS externally to use bused with these changes:
https://github.com/o3de/o3de/pull/3282

We have a 'Look Modification Component' in the PostFX system,
It allows the user to color grade the scene by loading a LUT (.3dl, as .azasset). We provide a few example luts.
HDR color grading (within ACES) has some specific needs, those are not documented.
We provide neither the docs or capabilities that help users generate their own, thus this PR
This PR adds some python scripts/utils related to color grading along with assets allowing for an 'artist friendly' color grading workflow/experience.

The workflow and python scripts may require OpenOmageOI, possibly the oiiotool.exe and python bindings, which are built in o3de \bin\profile:
"C:\Depot\o3de-engine\build\bin\profile\OpenImageIO.dll"
"C:\Depot\o3de-engine\build\bin\profile\oiiotool.exe"
"C:\Depot\o3de-engine\build\bin\profile\OpenImageIO.pyd"

Some of these scripts are meant to run in the editor, others are made available to the editor so we can build additional tools around them in the future.

The workflow also utilizes OpenColorIO (OCIO) and some of it's tools, currently o3de only builds the core ocio library (dll), so there are some other ways around this (for now, but a complete OCIO integration with the full tools set as well as the python bindings are what would be ideal)

I built OpenColorIO myself, we can provide instructions for that.
You can get most of the needed functionality either from OpenImageIO, or you can use the Photoshop ocio plugin (even though it is a bit clunky.)  We can also just build all of the color space and view transforms the workflow requires and provide those (that is an approach I have started with and include.)

Currently the verified and viable workflow is to do the color grading via photoshop (which is accessible to most game artists)
That approach will be outlined here (until we can get to writing official docs): #1963

We have an internal Amazon wiki for this workflow, I am updating/moving that information into the discussion linked above.

Signed-off-by: Jonny Galloway <gallowj@amazon.com>

### CODE REVIEW

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py
Comment (invertednormal) [inline Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py:151]: This could be simplified to
```
logMiddleGray = math.log(middleGrey, 2.0)
logMin = logMiddleGray + lowerStops
logMax = logMiddleGray + upperStops
```

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py
Comment (invertednormal) [inline Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py:131]: You could use math.log(x, 2.0) here, then replace calls below to that function to just log2(x)

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py
Comment (invertednormal) [inline Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py:233]: This could share most of the code with 3dl_to_azasset.py, maybe rip it out to lut_helper?

**Review by invertednormal — APPROVED** [review]
Looks good just a few suggestions for cleaning up some of the python code.

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/__init__.py
Comment (lawsonamzn) [inline Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/__init__.py:4]: this and a few others appear to be missing the line about looking for LICENSE at the root of the project

File: Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat
Comment (lawsonamzn) [inline Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat:8]: IS this a spelling mistake in the name of this batch file?  "ColorGradinTools"

File: Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat
Comment (lawsonamzn) [inline Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat:10]: ITs no longer LY, its o3De

File: Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/Env_Tools.bat
Comment (lawsonamzn) [inline Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/Env_Tools.bat:39]: Why does a vcpkg installed package live here, and not, eg, in a 3p package?  whats going on here?

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/3dl_to_azasset.py
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/3dl_to_azasset.py:56]: oh yeah, I actually encountered this when I tried to convert a photoshop LMT 3DL lut

File: Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/Env_Tools.bat
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/Env_Tools.bat:39]: I had to build the OCIO tools myself, and I used vcpkg to do it... so this is piped in here right now so I can carry on with my work.  I don't want to be blocked by a 3P concern that can be done separately.  I can either remove this before I merge these changes, or just update this in the future.  I need to write up a issue for fully adding/building OCIO and all of it's tools, right now we only build the OpenColorIO.dll that is in the engine bin.

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/__init__.py
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/__init__.py:4]: I will go figure out what is missed

File: Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Tools/ColorGrading/cmdline/CMD_ColorGradinTools.bat:8]: oops

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/from_3dl_to_azasset.py
Comment (invertednormal) [inline Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/from_3dl_to_azasset.py:42]: Based on how your appending intervals after the header, I think this } would need to be removed or all your interval numbers would come after the } and the whole json would be malformed. Another option would be to write the whole json string with entries for the intervals and values:
```
AZASSET_LUT = """\
{
    "Type": "JsonSerialization",
    "Version": 1,
    "ClassName": "LookupTableAsset",
    "ClassData": {
        "Name": "LookupTable",
        "Intervals": %s,
        "Values": %s
    }
}
```
Then you create your strings of values and intervals, and when you want to composite the whole file, just 
```
AZASSET_LUT % (intervals, values)
```
It'd trim this function down to like 4 or 5 lines.

File: Gems/Atom/Feature/Common/Tools/ColorGrading/Resources/TestData/Photoshop/HDR_in_LDR_Log2-48nits/LUT_PreShaped_Composite/hue-sat_inv-Log2-48nits_32_LUT.azasset
Comment (invertednormal) [inline Gems/Atom/Feature/Common/Tools/ColorGrading/Resources/TestData/Photoshop/HDR_in_LDR_Log2-48nits/LUT_PreShaped_Composite/hue-sat_inv-Log2-48nits_32_LUT.azasset]: Yeah - this is malformed json based on above feedback.

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/from_3dl_to_azasset.py
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/from_3dl_to_azasset.py:42]: That makes more sense, I'll also make it more py3 ready and use an f'string'

Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/from_3dl_to_azasset.py]: `def write_azasset(file_path, lut_size, lut_intervals, lut_values, azasset_json=AZASSET_LUT):

    values_str = ''
    for idx, px in enumerate(lut_values):
        values_str += "{0}, {1}, {2}\n".format(px[0], px[1], px[2])

    intervals_str = ', '.join(map(str, lut_intervals))
    
    azasset_json = azasset_json % (intervals_str, values_str)
    
    asset_file_path = f"{file_path}.azasset"
    _LOGGER.info(f"Writting {asset_file_path}...")
    asset_file = open(asset_file_path, 'w')
    asset_file.write(azasset_json)`

**Comment by lawsonamzn** [comment]
Please do a copyright header scrub - they all need to be the same and they all need to mention looking at the root for LICENSE (ie, the standard text)

File: Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/3dl_to_azasset.py
Comment (HogJonny-AMZN) [reply Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/3dl_to_azasset.py]: @invertednormal I think I resolved this

**Review by invertednormal — APPROVED** [review]
Looks good except old_lut_helper.py probably needs to be removed, and probably the Wing solution file as well.

**Review by invertednormal — APPROVED** [review]

**Review by galibzon — APPROVED** [review]

**Pull request merged**
